pennylane/interfaces/tensorflow.py

```diff
@@ -116,207 +165,110 @@ def _res_restructured(res, tapes):
     return tuple(res_nested)
 
 
-def _jac_restructured(jacs, tapes):
-    """
-    Reconstruct the nested tuple structure of the jacobian of a list of tapes
-    """
-    start = 0
-    jacs_nested = []
-    for tape in tapes:
-        num_meas = len(tape.measurements)
-        num_params = len(tape.trainable_params)
-
-        tape_jacs = tuple(jacs[start : start + num_meas * num_params])
-        tape_jacs = tuple(
-            tuple(tape_jacs[i * num_params : (i + 1) * num_params]) for i in range(num_meas)
-        )
-
-        while isinstance(tape_jacs, tuple) and len(tape_jacs) == 1:
-            tape_jacs = tape_jacs[0]
-
-        jacs_nested.append(tape_jacs)
-        start += num_meas * num_params
-
-    return tuple(jacs_nested)
-
-
-def _recursive_conj(dy):
-    if isinstance(dy, (tf.Variable, tf.Tensor)):
-        return tf.math.conj(dy)
-    return tuple(_recursive_conj(d) for d in dy)
-
-
-def execute(tapes, device, execute_fn, gradient_fn, gradient_kwargs, _n=1, max_diff=2):
+def tf_execute(tapes, execute_fn, jpc, device=None, differentiable=False):
     """Execute a batch of tapes with TensorFlow parameters on a device.
 
     Args:
         tapes (Sequence[.QuantumTape]): batch of tapes to execute
-        device (pennylane.Device): Device to use to execute the batch of tapes.
-            If the device does not provide a ``batch_execute`` method,
-            by default the tapes will be executed in serial.
-        execute_fn (callable): The execution function used to execute the tapes
-            during the forward pass. This function must return a tuple ``(results, jacobians)``.
-            If ``jacobians`` is an empty list, then ``gradient_fn`` is used to
-            compute the gradients during the backwards pass.
-        gradient_kwargs (dict): dictionary of keyword arguments to pass when
-            determining the gradients of tapes
-        gradient_fn (callable): the gradient function to use to compute quantum gradients
-        _n (int): a positive integer used to track nesting of derivatives, for example
-            if the nth-order derivative is requested.
-        max_diff (int): If ``gradient_fn`` is a gradient transform, this option specifies
-            the maximum number of derivatives to support. Increasing this value allows
-            for higher order derivatives to be extracted, at the cost of additional
-            (classical) computational overhead during the backwards pass.
+        execute_fn (Callable[[Sequence[.QuantumTape]], ResultBatch]): a function that turns a batch of circuits into results
+        jpc (JacobianProductCalculator): a class that can compute the vector Jacobian product (VJP)
+            for the input tapes.
+
+    Keyword Args:
+        device=None: not used for tensorflow
+        differentiable = False: whether or not the custom gradient vjp needs to be

```

Comment:
hopefully fixes rendering since it's the same as `device`
```suggestion
        differentiable=False: whether or not the custom gradient vjp needs to be
```